Activities/Cryptography/UiPath.Cryptography.Activities/Models/CryptographyLocalItem.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UiPath.Platform.ResourceHandling;
+
+namespace UiPath.Cryptography.Activities.Models
+{
+    /// <summary>
+    /// Used to convert cryptography files as IResources.
+    /// </summary>
+    public class CryptographyLocalItem : ILocalResource
+    {
+        /// <summary>
+        /// Create the ILocalResource implementation
+        /// </summary>
+        /// <param name="byteArray"></param>
+        /// <param name="fullName"></param>
+        internal CryptographyLocalItem(byte[] byteArray, string fullName)
+        {
+            var path = Path.GetRandomFileName();
+
+            File.WriteAllBytes(path, byteArray);
+
+            _fileInfo = new FileInfo(path);
+
+            FullName = !string.IsNullOrEmpty(fullName) ? fullName : _fileInfo.FullName;
+        }
+
+        #region Internal properties and fields
+
+        private readonly FileInfo _fileInfo;
+
+        #endregion Internal properties and fields
+
+        #region ILocalResource implementation
+
+        /// <inheritdoc />
+        public DateTime? CreationDate => _fileInfo?.CreationTime;
+
+        /// <inheritdoc />
+        public string FullName { get; }
+
+        /// <inheritdoc />
+        public string IconUri => string.Empty;
+
+        /// <inheritdoc />
+        public string ID => Guid.NewGuid().ToString();
```

Comment:
This will generate a new ID every time the property is accessed. Wrong on so many levels.